client/src/panels/panel_declarations/finances/auth_exp_planned_spending/auth_exp_planned_spending.js

```diff
@@ -261,65 +273,321 @@ class AuthExpPlannedSpendingGraph extends React.Component {
     );
   }
 }
+class LapseByVotesGraph extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      is_showing_lapse_pct: false,
+      active_votes: this.get_active_votes(
+        ({ votestattype }) =>
+          props.subject.level === "gov" ||
+          votestattype === 1 ||
+          votestattype === 2 ||
+          votestattype === 3
+      ),
+    };
+  }
+  get_active_votes = (func) =>
+    _.chain(this.props.queried_votes)
+      .map((vote_row) => [vote_row.desc, func(vote_row)])
+      .fromPairs()
+      .value();
+
+  render() {
+    const { subject, queried_votes, additional_info } = this.props;
+    const { active_votes, is_showing_lapse_pct } = this.state;
+    const filtered_votes = _.reject(
+      queried_votes,
+      ({ desc }) => !active_votes[desc]
+    );
+
+    const get_lapse_raw_data = (is_pct, votes = filtered_votes) =>
+      _.flatMap(votes, (vote_row) =>
+        _.map(std_years, (yr) =>
+          calculate_lapse(
+            vote_row[`${yr}auth`],
+            vote_row[`${yr}exp`],
+            vote_row[`${yr}unlapsed`],
+            is_pct
+          )
+        )
+      );
+    const lapsed_by_votes_sum = _.sum(get_lapse_raw_data(false, queried_votes));
+    const avg_lapsed_by_votes_pct = _.chain(queried_votes)
+      .reduce(
+        (result, vote_row) => ({
+          ..._.chain(flat_auth_exp_years)
+            .map((yr) => [yr, result[yr] + vote_row[yr]])
+            .fromPairs()
+            .value(),
+        }),
+        _.chain(flat_auth_exp_years)
+          .map((yr) => [yr, 0])
+          .fromPairs()
+          .value()
+      )
+      .thru((vote) =>
+        _.map(std_years, (yr) => {
+          const lapse_pct =
+            calculate_lapse(
+              vote[`${yr}auth`],
+              vote[`${yr}exp`],
+              vote[`${yr}unlapsed`]
+            ) / vote[`${yr}auth`];
+          return _.isNaN(lapse_pct) ? 0 : lapse_pct;
+        })
+      )
+      .mean()
+      .value();
+
+    const nivo_pct_props = is_showing_lapse_pct && {
+      is_money: false,
+      left_axis: { format: formats.smart_percentage2_raw },
+      text_formatter: formats.smart_percentage2,
+    };
+    const lapse_infograph = (
+      <div style={{ padding: "5px" }} className="frow">
+        <TM
+          className="medium-panel-text"
+          k={
+            subject.is("gov")
+              ? "gov_lapse_by_votes_text"
+              : "dept_lapse_by_votes_text"
+          }
+          args={{
+            subject,
+            avg_lapsed_by_votes: lapsed_by_votes_sum / std_years.length,
+            num_of_votes: queried_votes.length,
+            avg_lapsed_by_votes_pct,
+            gov_avg_lapsed_by_votes_pct:
+              additional_info.gov_avg_lapsed_by_votes_pct,
+          }}
+        />
+        <div
+          style={{ marginBottom: "10px" }}
+          className="fcol-md-12 medium-panel-text"
+        >
+          <div>
+            <input
```

Comment:
Did you consider the `RadioButtons` component? Might look a little fancier, up to you though (let me know if you preferred standard radio buttons for some reason).

Reply:
Didn't know we had RadioButtons component😂, Changed it, I think it looks better but kinda resemble the tabbed control. Oh well, I'm sticking with it.